pkg/apis/vsphere/v1/template_test.go

```diff
@@ -0,0 +1,94 @@
+package v1
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/ghttp"
+	"go.anx.io/go-anxcloud/pkg/api"
+	"go.anx.io/go-anxcloud/pkg/api/types"
+	corev1 "go.anx.io/go-anxcloud/pkg/apis/core/v1"
+	"go.anx.io/go-anxcloud/pkg/client"
+)
+
+var _ = Describe("template tests", func() {
+	Context("with mocked server", func() {
+		var a api.API
+		locationID := "mock-location-id"
+
+		BeforeEach(func() {
+			srv := ghttp.NewServer()
+			srv.AppendHandlers(ghttp.CombineHandlers(
+				ghttp.VerifyRequest("GET", fmt.Sprintf("/api/vsphere/v1/provisioning/templates.json/%s/templates", locationID)),
+				ghttp.RespondWithJSONEncoded(200, mockedTemplateList()),
+			))
+
+			var err error
+			a, err = api.NewAPI(api.WithClientOptions(
+				client.BaseURL(srv.URL()),
+				client.IgnoreMissingToken(),
+			))
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("can List templates", func() {
+			var channel types.ObjectChannel
+			err := a.List(context.TODO(), &Template{Type: TypeTemplate, Location: corev1.Location{Identifier: locationID}}, api.ObjectChannel(&channel))
+			Expect(err).ToNot(HaveOccurred())
+
+			templateCount := 0
+			template := &Template{}
+			for res := range channel {
+				err := res(template)
```

Comment:
```go
for res := range channel {
    t := Template{}
    err := res(&t)
```

at the very least, do not declare `template` as `*Template`, to match common usage